### PR TITLE
[core] revert error logging

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -1,14 +1,10 @@
 # Release History
 
-## 1.21.1 (Unreleased)
-
-### Features Added
-
-### Breaking Changes
+## 1.21.1 (2021-12-06)
 
 ### Bugs Fixed
```
Comment (xiangyan99) [inline sdk/core/azure-core/CHANGELOG.md:5]: Other changes

File: sdk/core/azure-core/CHANGELOG.md [hunk 2]
```diff
@@ -1,14 +1,10 @@
...
 
-### Other Changes
+- Revert exposure of response body in string representation of `HttpResponseError`  #22023
```
Comment (xiangyan99) [inline sdk/core/azure-core/CHANGELOG.md:7]: Revert the change of __str__ method

**Review by lmazuel — APPROVED** [review]
